Move method descriptions to docstrings

BenjaminBorn opened the pull request:
In this pull request, I have copied all available documentation on tests, pvalues, and confidence intervals to docstrings. Please note that I have only done some reformatting. I have not checked the documentation for correctness. This is the first step in the transition to Documenter.jl as discussed in #103. 

When reviewing, please consider the following questions.

1. I have attached the general description of `pvalue()` and `confint()` somewhat arbitrarily to two of the tests (`Binomial.jl` and `t.jl`). Is there a better place to put them?
2. Sometimes a test has different methods, e.g. `OneSampleTTest()`. If possible, I have combined them in one docstring, but sometimes, for clarity reasons, I have included different docstrings for the methods. Is that ok, or should we combine them somehow into one docstring? The disadvantage of having them in different docstrings is that the help output in the REPL can get convoluted, especially as there is not much spacing between the output for the different methods.
3. `pvalue` and `confint` also have different methods with a number of separate docstrings. Is there a way to ensure that the general method, e.g. `pvalue(x::HypothesisTests)`, is the first shown when typing `?pvalue`? As some of the more specialised methods have long docstrings, it could get lost otherwise.
3. The admonition `!!! note` doesn't work in the Jupyter notebook (at least not for me using Safari). It works in the REPL and Juno. Maybe someone can try it. If it's a general problem, we can file an issue with IJulia (or at the appropriate place).
4. There was a copy/paste mistake in the docstring for the binomial confidence intervals. After fixing it, I realised that there is also an open PR (#55) for it. Is there any way of giving credit for the fix to @juliangehring ?

Thanks in advance for your input!

BenjaminBorn commented:
One idea for question (1) would be to put a "fake" general method in `HypothesisTests.jl` à la
```
"""
    confint(test::HypothesisTest, alpha = 0.05; tail = :both)

Compute a confidence interval C with coverage 1-`alpha`.

...

"""
confint(test::HypothesisTest, alpha = 0.05; tail = :both)
```
And similarly for `pvalue`. Would that be preferable to attaching it to a random method?

ararslan commented:
A way to generally attach docstrings to functions is to either forward-declare the function with no methods, e.g.
```julia
"""
    confint(...)

...
"""
function confint end
```
or attach it to the function after methods have been defined for it, e.g.
```
"""
    confint(...)

...
"""
confint
```

nalimilan commented on src/fisher.jl:
```diff
@@ -36,11 +36,37 @@ immutable FisherExactTest <: HypothesisTest
     ω::Float64
+end
 
-    function FisherExactTest(a::Int, b::Int, c::Int, d::Int)
```
Shouldn't move this function out of the type definition: this would make it an outer constructor rather than an inner constructor. Just attach the docstring to the type itself.

nalimilan commented on src/fisher.jl:
```diff
@@ -36,11 +36,37 @@ immutable FisherExactTest <: HypothesisTest
+|*Y2*| c  | d  |
+
+!!! note
+    The print output contains the conditional maximum likelihood estimate of the odd-ratio
```
"odds ratio" (also fix other occurrences below). Also, "print" would better be "show", with a link to the function.

nalimilan commented on src/fisher.jl:
```diff
@@ -59,6 +85,41 @@ function show_params(io::IO, x::FisherExactTest, ident="")
+Compute the p-value for a given significance test.
+
+The one-sided p-values are based on Fisher's non-central hypergeometric distribution
+``f_\\omega(i)`` with odd-ratio ``\\omega``:
```
You can use the omega character directly. Same for `leq` and `geq`.

nalimilan commented on src/fisher.jl:
```diff
@@ -59,6 +85,41 @@ function show_params(io::IO, x::FisherExactTest, ident="")
+"""
+    pvalue(x::FisherExactTest; tail = :both, method = :central)
+
+Compute the p-value for a given significance test.
```
"for a given Fisher exact test".

nalimilan commented on src/fisher.jl:
```diff
@@ -36,11 +36,37 @@ immutable FisherExactTest <: HypothesisTest
+# References
+
+  * Fay, M.P. Supplementary material to confidence intervals that match Fisher’s exact or
+    Blaker’s exact tests. Biostatistics, 0(0): 1-13, 2009.
```
"0(0)" is suspicious. Also below.

nalimilan commented on src/fisher.jl:
```diff
@@ -59,6 +85,41 @@ function show_params(io::IO, x::FisherExactTest, ident="")
+
+# References
+
+  * Gibbons, J.D, Pratt, J.W. P-values: Interpretation and Methodology American
```
Missing dot before review name.

nalimilan commented on src/fisher.jl:
```diff
@@ -97,6 +158,18 @@ function pvalue_both_minlike(x::FisherExactTest, ω::Float64=1.0)
+"""
+    confint(x::FisherExactTest, alpha::Float64=0.05; tail=:both, method=:central)
+
+Compute a confidence interval with coverage 1 - `alpha` by inverting the `:central` p-value.
```
Would be nice to repeat the definition of different methods here. AFAIK there's no need to mention the p-value here at all.

nalimilan commented on src/fisher.jl:
````diff
@@ -59,6 +85,41 @@ function show_params(io::IO, x::FisherExactTest, ident="")
+    ```
+
+!!! note
+    Since the p-value is not necessarily unimodal, the corresponding confidence region might
````
Why mention this here and not in the docstring for `confint`?

nalimilan commented on src/kolmogorov_smirnov.jl:
```diff
@@ -54,6 +54,15 @@ immutable ExactOneSampleKSTest <: ExactKSTest
+"""
+    ExactOneSampleKSTest{T<:Real}(x::AbstractVector{T}, d::UnivariateDistribution)
+
+Perform a one sample Kolmogorov–Smirnov-test of the null hypothesis that the data in vector
```
"Kolmogorov–Smirnov test". Same below.

Would also make sense to mention "exact".

nalimilan commented on src/kolmogorov_smirnov.jl:
```diff
@@ -129,6 +147,21 @@ immutable ApproximateTwoSampleKSTest <: ApproximateKSTest
+
+Perform an asymptotic two sample Kolmogorov–Smirnov-test of the null hypothesis that `x`
+and `y` are drawn from the same distribution against the alternative hypothesis that the
+distribution comes from different distributions.
```
"distribution comes" -> "they come"

nalimilan commented on src/kolmogorov_smirnov.jl:
```diff
@@ -129,6 +147,21 @@ immutable ApproximateTwoSampleKSTest <: ApproximateKSTest
+# External links
+
+  * [Approximation of one-sided test:
+     https://www.encyclopediaofmath.org/index.php/Kolmogorov-Smirnov_test](
```
Better give just the full name of the site rather than repeating the URL.

nalimilan commented on src/kruskal_wallis.jl:
```diff
@@ -32,6 +32,35 @@ immutable KruskalWallisTest <: HypothesisTest
+"""
+    KruskalWallisTest{T<:Real}(groups::AbstractVector{T}...)
+
+Perform Kruskal Wallis rank sum test of the null hypothesis that the location parameters of
```
"Kruskal-Wallis"

nalimilan commented on src/kruskal_wallis.jl:
```diff
@@ -32,6 +32,35 @@ immutable KruskalWallisTest <: HypothesisTest
+    KruskalWallisTest{T<:Real}(groups::AbstractVector{T}...)
+
+Perform Kruskal Wallis rank sum test of the null hypothesis that the location parameters of
+the distribution of the ``n`` observations are the same in each of the `groups`
```
It's not clear what `n` corresponds to.

nalimilan commented on src/kruskal_wallis.jl:
```diff
@@ -32,6 +32,35 @@ immutable KruskalWallisTest <: HypothesisTest
+
+Perform Kruskal Wallis rank sum test of the null hypothesis that the location parameters of
+the distribution of the ``n`` observations are the same in each of the `groups`
+``\\mathcal{G}`` against the alternative hypothesis that they differ in at least one.
```
"at least one" sounds weird to me. I can't find this description in other references.

BenjaminBorn replied:
At least one group has to stochastically dominate one other group.

nalimilan commented on src/mann_whitney.jl:
```diff
@@ -52,7 +71,17 @@ end
+"""
+    ExactMannWhitneyUTest(x::AbstractVector{T<:Real}, y::AbstractVector{T<:Real})
 
+Perform an exact Mann-Whitney U test.
```
Copy the definition of this test from the above docstring, as it should be usable in isolation. Same for the approximate test.

nalimilan commented on src/mann_whitney.jl:
```diff
@@ -52,7 +71,17 @@ end
 
 
 ## EXACT MANN-WHITNEY U TEST
+"""
```
Better attach this docstring to the constructor than to the type since it documents the function. Same below.

nalimilan commented on src/power_divergence.jl:
```diff
@@ -44,6 +43,28 @@ default_tail(test::PowerDivergenceTest) = :right
+Compute a confidence interval with coverage 1-`alpha` for multinomial proportions using
+one of the following methods. Possible values for `method` are:
+
+  - Sison, Glaz intervals `:sison_glaz` (default)
```
Would be great if you could describe these a bit.

nalimilan commented on src/power_divergence.jl:
```diff
@@ -44,6 +43,28 @@ default_tail(test::PowerDivergenceTest) = :right
+    90:366-369, 1995.
+  * Quesensberry, C.P. and Hurst, D.C. Large Sample Simultaneous Confidence Intervals for
+    Multinational Proportions. Technometrics, 6:191-195, 1964.
+  * Gold, R. Z. Tests Auxiliary to :math:`{\chi^{2}}` Tests in a Markov Chain. Annals of
```
You can write  χ² directly.

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
 # Under regularity conditions, their asymptotic distributions are all the same (Drost 1989)
 # Chi-squared null approximation works best for lambda near 2/3
+"""
+    PowerDivergenceTest(x [,y] [, lambda] [,theta0] )
```
Spaces must go after commas. Also mention the default values of `lambda` and `theta`. It could make sense to use two lines for the signature, one without `y` and one with it.

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+Perform a Power Divergence Test.
+
+If `x` is a matrix with one row or column, or if `x` is a vector and `y` is not given, then
+a goodness-of-fit test is performed (``x`` is treated as a one-dimensional contingency
```
Single backticks around `x`. Missing closing parenthesis.

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+`theta0` is not given.
+
+If `x` is a matrix with at least two rows and columns, it is taken as a two-dimensional
+contigency table: the entries of `x` must be non-negative integers. Otherwise, `x` and
```
"contingency". It seems that "the entries of `x` must be non-negative integers" applies all the time, and also to `y` when it is passed, so maybe better mention this at the end in a separate paragraph?

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+
+Perform a Power Divergence Test.
+
+If `x` is a matrix with one row or column, or if `x` is a vector and `y` is not given, then
```
It's not clear whether "`y` is not given" applies to both cases or only to the first one.

BenjaminBorn replied:
I think it applies to both cases and have reformulated accordingly.

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+If `x` is a matrix with at least two rows and columns, it is taken as a two-dimensional
+contigency table: the entries of `x` must be non-negative integers. Otherwise, `x` and
+`y` must be vectors of the same length. The contigency table is calculated using `counts`
+from `Statsbase`. Then the power divergence test is performed of the null hypothesis that
```
Don't mention the package, just use `@ref`.

nalimilan commented on src/power_divergence.jl:
````diff
@@ -214,6 +235,48 @@ end
+    /\\hat{n}_{ij})^\\lambda -1\\right]
+```
+where ``n_{ij}`` is the cell count in the ``i`` th row and ``j`` th column and ``\\lambda``
+is a real number that has the following feature
````
"is a real number that has the following feature" -> "is a real number determing the nature of the test to be performed:".

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+  * ``λ = -2``: it equals Neyman modified chi-squared (Neyman 1949)
+  * ``λ = -1/2``:  it equals the Freeman-Tukey statistic (Freeman and Tukey 1950).
+
+Under regularity conditions, the asymptotic distributions are identical (see Drost et. al.
```
"asymptotic distributions" of what?

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+  * ``λ = -1/2``:  it equals the Freeman-Tukey statistic (Freeman and Tukey 1950).
+
+Under regularity conditions, the asymptotic distributions are identical (see Drost et. al.
+1989). The chi-squared null approximation works best for ``λ`` near ``2/3``.
```
What's the "the chi-squared null approximation" in the list above?

nalimilan commented on src/power_divergence.jl:
```diff
@@ -299,6 +367,11 @@ ChisqTest{T<:Integer,U<:AbstractFloat}(x::AbstractVector{T}, theta0::Vector{U} =
+"""
+    MultinomialLRT(x [,y] [,theta0])
+
+Convenience function for power divergence test with ``λ=0``.
```
Explain more what this test means so that it can be used in isolation. Link to `PowerDivergenceTest`.

nalimilan replied:
Please at least copy the relevant docs from `PowerDivergenceTest` and fix the signature.

nalimilan commented on src/t.jl:
```diff
@@ -33,6 +33,24 @@ pvalue(x::TTest; tail=:both) = pvalue(TDist(x.df), x.t; tail=tail)
+    Most of the implemented confidence intervals are *strongly consistent*, that is, the
+    confidence interval with coverage 1-`alpha` does not contain the test statistic under
+    ``h_0`` if and only if the corresponding test rejects the null hypothesis
+    ``h_0: \\theta=\\theta_0``:
```
You can write alpha and theta directly using greek letters.

nalimilan commented on src/t.jl:
```diff
@@ -108,6 +153,15 @@ end
+    EqualVarianceTTest(x::AbstractVector{T<:Real}, y::AbstractVector{T<:Real})
+
+Perform a two-sample t-test of the null hypothesis that `x` and `y` come from a
+distributions with the same mean and equal variances against the alternative hypothesis
```
"a distributions  with the same mean and equal variances" -> "distributions with equal means and variances". Same below.

nalimilan commented on src/wilcoxon.jl:
```diff
@@ -27,6 +27,21 @@ export SignedRankTest, ExactSignedRankTest, ApproximateSignedRankTest
+When there are no tied ranks and ≤50 samples, or tied ranks and ≤15 samples,
+`SignedRankTest` performs an exact signed rank test. In all other cases, `SignedRankTest`
+performs an approximate signed rank test. Behavior may be further controlled by using
+`ExactSignedRankTest` or `ApproximateSignedRankTest` directly.
```
Use `@ref`.

nalimilan commented on src/mann_whitney.jl:
```diff
@@ -27,6 +27,25 @@ export MannWhitneyUTest, ExactMannWhitneyUTest, ApproximateMannWhitneyUTest
+
+The Mann-Whitney U test is sometimes known as the Wilcoxon rank sum test.
+
+When there are no tied ranks and ≤50 samples, or tied ranks and ≤10 samples,
```
Use `@ref`.

nalimilan commented on src/wilcoxon.jl:
```diff
@@ -62,6 +77,17 @@ immutable ExactSignedRankTest{T<:Real} <: HypothesisTest
+"""
+    ExactSignedRankTest(x::AbstractVector{T<:Real}[, y::AbstractVector{T<:Real}])
+
+Perform an exact signed rank U test.
```
Repeat the definition of the test. Same below.

nalimilan commented:
> I have attached the general description of pvalue() and confint() somewhat arbitrarily to two of the tests (Binomial.jl and t.jl). Is there a better place to put them?

The usual solution to that is to define `function pvalue end` somewhere and attach the generic docstring to that.

> Sometimes a test has different methods, e.g. OneSampleTTest(). If possible, I have combined them in one docstring, but sometimes, for clarity reasons, I have included different docstrings for the methods. Is that ok, or should we combine them somehow into one docstring? The disadvantage of having them in different docstrings is that the help output in the REPL can get convoluted, especially as there is not much spacing between the output for the different methods.

I'd say that's OK as long as the repeated content is not too long. If the REPL output isn't clear, it should be improved, rather than working around it in packages (the HTML manual looks better in general).

> pvalue and confint also have different methods with a number of separate docstrings. Is there a way to ensure that the general method, e.g. pvalue(x::HypothesisTests), is the first shown when typing ?pvalue? As some of the more specialised methods have long docstrings, it could get lost otherwise.

IIRC it depends on the order in which the methods are defined. Using the trick I gave above, you should be able to define it in the right place.

> The admonition !!! note doesn't work in the Jupyter notebook (at least not for me using Safari). It works in the REPL and Juno. Maybe someone can try it. If it's a general problem, we can file an issue with IJulia (or at the appropriate place).

Agreed. Also check that it works in the HTML manual though.

> There was a copy/paste mistake in the docstring for the binomial confidence intervals. After fixing it, I realised that there is also an open PR (#55) for it. Is there any way of giving credit for the fix to @juliangehring ?

Yes, the best way to give her credit is to merge here PR, which I just did. Thanks for pointing it out!

BenjaminBorn commented on src/anderson_darling.jl:
```diff
@@ -30,6 +30,15 @@ immutable OneSampleADTest <: ADTest
 end
 
+"""
+    OneSampleADTest{T<:Real}(x::AbstractVector{T}, d::UnivariateDistribution)
```
Only in the docstring or should I also go through the function definitions?

nalimilan replied:
You could do both if `T` isn't used in the body of functions, but to keep this PR focused better only touch docstrings.

BenjaminBorn commented:
Ok, I have (hopefully) addressed all points directly related to the PR. I will now make a pass through the general doc improvement comments and commit them separately, however some specific questions concerning the tests are outside my expertise. Should I just leave them open?

BenjaminBorn commented on src/binomial.jl:
```diff
@@ -53,20 +66,52 @@ function show_params(io::IO, x::BinomialTest, ident="")
+  * Brown, L.D., Cai, T.T., and DasGupta, A. Interval estimation for a binomial proportion.
+    Statistical Science, 16(2):101–117, 2001.
+
+# External links
```
I would prefer to keep the links in a separate category for consistency reasons. That's how I have also done it for the time-series tests.

BenjaminBorn commented on src/binomial.jl:
```diff
@@ -53,20 +66,52 @@ function show_params(io::IO, x::BinomialTest, ident="")
+  - Agresti Coull interval `:agresti_coull`: Simplified version of the Wilson interval;
+    they are centered around the same value. The Agresti Coull interval has higher or
+    equal coverage.
+  - Arcsine transformation `:arcsine`.
```
I have said a word about it ;-) However, I don't know how well they do compared to the other intervals.

nalimilan replied:
OK, let's leave this for somebody else then. ;-)

ararslan commented on src/kruskal_wallis.jl:
````diff
@@ -32,6 +32,41 @@ immutable KruskalWallisTest <: HypothesisTest
+The p-value is computed using a ``χ^2`` approximation to the distribution of the test
+statistic ``H_c=\\frac{H}{C}``:
+```math
+    \\begin{align}
````
I forget, does `align` number the lines and `align*` doesn't?

BenjaminBorn replied:
Good point, `align` numbers (although the Jupyter notebook doesn't print them). I'll change it to `align*` in the next iteration.

ararslan commented:
```@docs
    pvalue
    ```

🙂

BenjaminBorn commented:
````
```@docs
pvalue
```
````
But that prints the docstrings for *all* defined methods of `pvalue`. What if I only want the general one? For example so that I can structure the documentation and put `pvalue(::Binomial)` somewhere else.

ararslan commented:
Ah, I see. I _think_ you should be able to do this:
```julia
"""
    pvalue(...)

Some general description
"""
pvalue(::HypothesisTest; kwargs...)
```
after methods have been defined for `pvalue`, rather than the 0-method forward declaration.

BenjaminBorn commented:
Thanks, but even if I put the definition at the end of `HypothesisTests.jl` after everything has been defined (and the docstring can be found, e.g. in the REPL), I still get
> ERROR: LoadError: UndefVarError: HypothesisTest not defined
while loading /Users/bborn/.julia/v0.6/HypothesisTests/docs/make.jl, in expression starting on line 3

when I add `pvalue(::HypothesisTest)` to the `@docs` block.

ararslan commented:
Hmmmmm, maybe

    ```@meta
    CurrentModule = HypothesisTests
    ```

    ```@docs
    pvalue(::HypothesisTest)
    ```

nalimilan commented on src/t.jl:
```diff
@@ -70,15 +70,42 @@ function show_params(io::IO, x::OneSampleTTest, ident="")
+    OneSampleTTest(xbar::Real, stddev::Real, n::Int, μ0::Real = 0)
+
+Perform a one sample t-test of the null hypothesis that `n` values with mean `xbar` and
+sample standard deviation `stddev`  come from a distribution with `μ0` against the
```
Still need to fix "with mean `μ0`".

nalimilan commented on src/power_divergence.jl:
```diff
@@ -214,6 +235,48 @@ end
+
+If `x` is a matrix with at least two rows and columns, it is taken as a two-dimensional
+contingency table. Otherwise, `x` and `y` must be vectors of the same length. The contingency
+table is calculated using `counts` from [`Statsbase`](@ref). Then the power divergence test
```
"StatsBase". Also, I meant writing``[`StatsBase.counts`](@ref)`` (I'm not sure a link to a module works).

BenjaminBorn replied:
The `Documenter.jl` documentation states

> Note that depending on what the CurrentModule is set to, a docstring @ref may need to be prefixed by the module which defines it.

However, at least in my local build, links like ``[`StatsBase.counts`](@ref)`` don't seem to work. But maybe we can figure this out in the next PR that will actually set up `Documenter.jl`.

nalimilan commented on src/power_divergence.jl:
````diff
@@ -214,6 +235,48 @@ end
+    /\\hat{n}_{ij})^λ -1\\right]
+```
+where ``n_{ij}`` is the cell count in the ``i`` th row and ``j`` th column and ``λ`` is a
+real number determing the nature of the test to be performed:
````
"determining"

BenjaminBorn commented:
Thanks @ararslan and @nalimilan for the thorough review. I have hopefully addressed all points now. Once this PR is merged, I will open a new PR to build the docs.

nalimilan commented on src/fisher.jl:
```diff
@@ -71,7 +71,8 @@ immutable FisherExactTest <: HypothesisTest
 
 testname(::FisherExactTest) = "Fisher's exact test"
-population_param_of_interest(x::FisherExactTest) = ("Odds ratio", 1.0, x.ω) # parameter of interest: name, value under h0, point estimate
+population_param_of_interest(x::FisherExactTest) = ("Odds ratio", 1.0, x.ω)
```
Better not change this since that's unrelated. Or at least put the comment before the function rather than after. Same below.

BenjaminBorn replied:
Ok. I have reverted it.

BenjaminBorn commented:
Done. Let me know if I should do something else.

bjarthur commented:
this is great!  but...  why is this package not being tested on julia 0.6?

ararslan commented:
Will be. Once this is merged we should drop 0.5 support, test on 0.6, and have FemtoCleaner do a deprecation fixing pass.

ararslan commented on src/fisher.jl:
````diff
@@ -59,6 +86,37 @@ function show_params(io::IO, x::FisherExactTest, ident="")
+The one-sided p-values are based on Fisher's non-central hypergeometric distribution
+``f_ω(i)`` with odds ratio ``ω``:
+```math
+    \\begin{align}
````
I guess if you're moving everything to `align*` this should be as well? Aside from that, this all LGTM.

BenjaminBorn replied:
Thanks, fixed.

ararslan commented on src/fisher.jl:
```diff
@@ -24,6 +24,33 @@
+|*Y2*| c  | d  |
+
+!!! note
+    The [`Base.show`](@ref) output contains the conditional maximum likelihood estimate of the odds ratio
```
When you build the docs, what does this end up linking to?

nalimilan replied:
Unfortunately it seems it's not possible at the moment: https://github.com/JuliaDocs/Documenter.jl/issues/425. Let's go without links for now.

BenjaminBorn commented on src/fisher.jl:
```diff
@@ -24,6 +24,33 @@
+|*Y2*| c  | d  |
+
+!!! note
+    The [`Base.show`](@ref) output contains the conditional maximum likelihood estimate of the odds ratio
```
As noted above

> The Documenter.jl documentation states
"*Note that depending on what the CurrentModule is set to, a docstring @ref may need to be prefixed by the module which defines it.*"
> However, at least in my local build, links like [`StatsBase.counts`](@ref) don't seem to work. But maybe we can figure this out in the next PR that will actually set up Documenter.jl.

So, at least for now the links to outside modules don't work. Those are:

- ``[`StatsBase.counts`](@ref)`` and  `` [`Base.show`](@ref)`` for which the error is that `!! No doc found for reference`
- ``[`Rmath.pwilcox`](@ref)`` and ``[`Rmath.psignrank`](@ref)`` for which it says `ERROR: UndefVarError(:Rmath)`, so this is maybe not the correct module call.

The Documenter.jl documentation unfortunately doesn't give an example how to link to other modules and I also couldn't find an example in other packages.

Besides these dead links, the docs are still built correctly and everything looks ok. If you have an idea on how to fix it, I can push the fix here. If it takes us longer to figure it out, we might want to merge here and then fix the links in the follow-up PR.

ararslan replied:
I'd just remove the dead ref links for now, e.g. in this case just say "The `show` output contains..."

BenjaminBorn replied:
Done. But you explain this to @nalimilan ;-)

ararslan replied:
Did Milan tell you to link to things that can't be properly linked...?

BenjaminBorn replied:
No no, of course not. I was joking. He told me to include the links but wasn't sure whether it would work. Let's leave them out and we can figure the linking out later.

BenjaminBorn commented:
Thanks Alex. Sorry for the back and forth.

ararslan commented:
No problem at all! That's just how PR reviews go. You've done awesome work here, as usual.

ararslan commented:
@nalimilan Good to merge? Any further cleanup that's necessary can always happen later.

nalimilan commented:
Sorry for the delay. It's fine with me! Though conflicts have to be fixed first.

ararslan commented:
> Though conflicts have to be fixed first.

Weird, I'm not seeing any merge conflicts?
